service_configuration_lib/spark_config.py

```diff
@@ -291,31 +291,38 @@ def get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
     ):
         return spark_opts
 
+    spark_app_name = spark_opts.get('spark.app.name', '')
+
     log.warning(
-        'Spark Dynamic Resource Allocation (DRA) enabled for this batch. More info: y/spark-dra. '
-        'If your job is performing worse because of DRA, consider disabling DRA. To disable, '
-        'please provide spark.dynamicAllocation.enabled=false in your spark args\n',
+        'Spark Dynamic Resource Allocation (DRA) enabled for this batch. More info: y/spark-dra.\n',
     )
+    # TODO: add regex to better match Jupyterhub Spark session app name
+    if 'jupyterhub' not in spark_app_name:
+        log.warning(
+            'If your job is performing worse because of DRA, consider disabling DRA. To disable, '
+            'please provide spark.dynamicAllocation.enabled=false in your spark args\n',
+        )
 
-    spark_app_name = spark_opts.get('spark.app.name', '')
     # set defaults if not provided already
     _append_spark_config(spark_opts, 'spark.dynamicAllocation.enabled', 'true')
     _append_spark_config(spark_opts, 'spark.dynamicAllocation.shuffleTracking.enabled', 'true')
     _append_spark_config(
         spark_opts, 'spark.dynamicAllocation.executorAllocationRatio',
         str(DEFAULT_DRA_EXECUTOR_ALLOCATION_RATIO),
     )
+    if 'spark.dynamicAllocation.cachedExecutorIdleTimeout' not in spark_opts:
```

Comment:
Cached executor idle timeout differs for different use-cases.
Jupyter notebook usage tends for longer time-gaps between two executive actions. 
It is better not clean cache for longer times
We have seen that increasing cached timeout for jupyter notebooks has helped in running their jobs faster without recomputation.

I suggest the following:
* Have a private api to identify whether a job is coming from Jupyterhub framework or not.
* Even better having an api which sets spark configs for Jupyterhub framework use-cases (e.g. min executors as 0, timeout as 45min+), log cost for jupyterhub notebooks
* Set 30 min cached timeout for regular tron jobs
* Set 45 min or higher cached timeout for jupyterhub jobs (with message of timeout value being set)